schema/query/predicate_parser.go

```diff
@@ -258,6 +254,16 @@ func (p *predicateParser) parseCommand(field string) (Expression, error) {
 				return nil, fmt.Errorf("%s: expected '}' got %q", label, p.peek())
 			}
 			return &ElemMatch{Field: field, Exps: exps}, nil
+		case opNot:
+			re, err := p.parseRegex(label)
+			if err != nil {
+				return nil, err
+			}
+			p.eatWhitespaces()
+			if !p.expect('}') {
+				return nil, fmt.Errorf("%s: expected '}' got %q", label, p.peek())
+			}
+			return &Not{Field: field, Value: re}, nil
```

Comment:
Maybe we could instead return `$Regex{Field: field, Value: re, Negated: true}`.

This would give a clear path-way for how to support the `$not` operator in general later, if there is a need for that.

Reply:
great suggestion, thanks